utility/validator/module.go

```diff
@@ -0,0 +1,75 @@
+package validator
+
+import (
+	"github.com/pokt-network/pocket/logger"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+const (
+	ValidatorModuleName = "validator"
+)
+
+type ValidatorModule interface {
```

Comment:
I see the approach you took here. Still thinking about, but I do think it's a good start!